fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -3801,6 +3822,8 @@ public ChangedTransactionDetail closeAsWrittenOff(final JsonCommand command, fin
 
         validateAccountStatus(LoanEvent.WRITE_OFF_OUTSTANDING);
 
+        final LocalDate writtenOffOnLocalDate = command.localDateValueOfParameterNamed(TRANSACTION_DATE);
+        this.closedOnDate = writtenOffOnLocalDate;
```

Comment:
Why to move these dates?

Reply:
I needed to move the set of the `closedOnDate` value before the Life cycle machine method call because when the event listener was called It was in null at that moment